Automattic/wp-calypso
# Hosting Overview Refinements: Drop `data-scroll-id` in favor of refs

## Proposed Changes

Follow-up to #93223 and #93419.

This PR moves the logic for automatically scrolling to the WP / PHP version selectors from the topmost `Hosting` component down to `WebServerSettingsCard`. The benefit is that we don't need to rely on `setInterval` (an imprecise method), but we can use React refs.

## Testing Instructions



1. Navigate to `/sites?flags=hosting-overview-refinements`
2. Click on an Atomic site
3. Click on the WordPress version number in the header
4. Ensure that the page automatically scrolls down to the WP version picker after a second or two
5. Navigate back in your browser
6. Repeat the process for the PHP version picker
